src/seed_data/factories/Course.factory.ts

```diff
@@ -0,0 +1,43 @@
+import { define } from 'typeorm-seeding';
+import { Course } from '../../entity/Course';
+import { subjects, getRandomIntInclusive, toUpperCase } from '../utils'
+
+import * as faker from 'faker';
+
+faker.seed(123);
+
+let courseNumbers = [];
+
+function generateCourseNumber() {
+    let num = faker.random.number();
+    if (num < 1000) {
+        num *= 1000;
+    }
+
+    let stringNum = num.toString().substring(0, 4);
+
+    while (stringNum in courseNumbers) {
```

Comment:
im curious if the `unique` api call can be used here instead of the while loop checks according to this [kinda old github issue ](https://github.com/Marak/faker.js/issues/466#issuecomment-336057596).

`faker` doesn't seem to have a ton of documentation, which makes me sad :( 
so no worries if this isn't possible!

Reply:
hmm on second thought, disregard the comment above. it's prob not possible

Reply:
yea, I tried it out -- it would work if we didn't need arguments, but I think the comments above about avoid string parsing are of higher priority, so this won't be possible with the `faker` library